staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -680,7 +687,13 @@ func (alloc *allocator) allocateOne(r deviceIndices, allocateSubRequest bool) (b
 	claim := alloc.claimsToAllocate[r.claimIndex]
 	if r.requestIndex >= len(claim.Spec.Devices.Requests) {
 		// Done with the claim, continue with the next one.
-		return alloc.allocateOne(deviceIndices{claimIndex: r.claimIndex + 1}, false)
+		success, err := alloc.allocateOne(deviceIndices{claimIndex: r.claimIndex + 1}, false)
+		if errors.Is(err, errAllocationResultMaxSizeExceeded) {
+			// We don't need to propagate this further, since the allocation size
+			// limit is per claim.
+			return success, nil
```

Comment:
`success` is always false here, right?

`return false, nil` might make the code a bit more obvious.